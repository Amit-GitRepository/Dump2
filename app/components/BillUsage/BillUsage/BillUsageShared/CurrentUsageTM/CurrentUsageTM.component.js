import HotExtraPackageCard from '../HotExtraPackageCard/HotExtraPackageCard.component';
import Proptypes from 'prop-types';
import React, {Component} from 'react';
import styles from './CurrentUsageTM.style';
import {Button, Card, CircularProgress, Icon, ISText, List, Modal, Touchable} from '../../../../Shared';
import {bytesToSize} from '../../../../../utils/common.util';
import {colors} from '../../../../../themes/constants.styles';
import {isEmpty, noop} from 'lodash';
import {translate} from '../../../../../language/i18n/helper';
import {View} from 'react-native';

class CurrentUsageTM extends Component {

  state = {
    isVisible: false,
    modalData: [],
    modalHeaderTextKey: ''
  };

  toggleModal = (modalData = [], modalHeaderTextKey = '') => () => {
    const {isVisible} = this.state;
    this.setState({isVisible: !isVisible, modalData, modalHeaderTextKey});
  }

  getConsumedDataContent = (value, titleText, initialValue, unit, isPrepaid, isUnlimited) => {
    const footerText = isPrepaid ? unit : `${translate('BILLS_USAGE_OF')} ${initialValue} ${unit}`;
    const displayValue = (isUnlimited) ? translate('BILLS_USAGE_UNLIMITED') : value;
    return (
      <View style={ isPrepaid ? [styles.detailsContainer, styles.prepaidContainer] : styles.detailsContainer}>
        <ISText type='BOLD' style={styles.infoTopText}>{titleText}</ISText>
        <ISText style={styles.infoSubText}>{`(${translate('BILLS_USAGE_MAIN')} + ${translate('BILLS_USAGE_EXTRA')})`}</ISText>
        <ISText type='MEDIUM' style={[styles.valueText, styles.circularProgressFillStyle]}>{displayValue}</ISText>
        { isUnlimited ? null : <ISText style={styles.infoBottomText}>{footerText}</ISText>}
      </View>
    );
  }

  render () {
    const {hotExtraPackage, sharedNumbers, multiSimNumbers, navigateToBuyExtraPackage, isPrepaid, usageData} = this.props;
    const {isVisible, modalData, modalHeaderTextKey} = this.state;

    const {aggregatedTotalVoiceQuota, aggregatedRemainingVoiceQuota, isUnlimitedVoice, aggregatedTotalDataQuota,
      totalDataQuotaUnit, aggregatedRemainingDataQuota, isUnlimitedData, totalRemainingDataUnit} = usageData.AGGREGATED_QUOTA;

    let dataValue = aggregatedRemainingDataQuota;
    let dataUnit = totalRemainingDataUnit;
    if (!isPrepaid) { // For prepaid: Show remaining and for postpaid show used
      if (totalDataQuotaUnit !== totalRemainingDataUnit) { // Convert Remaining value to the same unit as total value
        const remainingData = bytesToSize(dataValue, totalRemainingDataUnit, totalDataQuotaUnit);
        dataValue = remainingData.value;
        dataUnit = totalDataQuotaUnit;
      }
      dataValue = (aggregatedTotalDataQuota - dataValue).toFixed(1);
    }

    const unlimitedData = isUnlimitedData && !aggregatedRemainingDataQuota && !aggregatedTotalDataQuota; // Display data value only if no total value
    const voiceValue = (isPrepaid) ? aggregatedRemainingVoiceQuota : aggregatedTotalVoiceQuota - aggregatedRemainingVoiceQuota;
    const voicePercentageConsumed = isUnlimitedVoice || isPrepaid ? 0 : (voiceValue * 100) / aggregatedTotalVoiceQuota;
    const dataPercentageConsumed = unlimitedData || isPrepaid ? 0 : (dataValue * 100) / aggregatedTotalDataQuota;

    const hasData = aggregatedTotalDataQuota || unlimitedData;
    const hasVoice = aggregatedTotalVoiceQuota || isUnlimitedVoice;
    const hasPackages = hasData || hasVoice;
    const hasOneOfDataVoice = hasData && !hasVoice || !hasData && hasVoice;
    const hasHotExtraPackage = !isEmpty(hotExtraPackage);
    const hasMultiSimPlan = !isEmpty(multiSimNumbers);
    const hasSharedPlan = !isEmpty(sharedNumbers);
    let hotExtraPackageCaretPosition = null; // means dont show caret

    if (hasHotExtraPackage && hasPackages) {
      if (hasOneOfDataVoice) {
        hotExtraPackageCaretPosition = 'CENTER';
      } else {
        switch (hotExtraPackage.planType) {
        case 'voice': { hotExtraPackageCaretPosition = 'LEFT'; break; }
        case 'data': { hotExtraPackageCaretPosition = 'RIGHT'; break; }
        }
      }
    }
    return (
      <View style={styles.currentUsageContainer}>
        { hasMultiSimPlan ?
          <Touchable onPress={this.toggleModal(multiSimNumbers, 'BILLS_USAGE_MULTI_SIM')}>
            <View style={styles.multiSimContainer}>
              <ISText type='BOLD' style={styles.multiSimText}>{`${translate('BILLS_USAGE_MULTI_SIM')} (${multiSimNumbers.length})`}</ISText>
            </View>
          </Touchable> : null
        }
        <View style={styles.currentUsageWrapper}>
          {
            hasPackages ?
              <View style={styles.currentUsageSubContainer}>
                {
                  hasVoice ?
                    isUnlimitedVoice || isPrepaid ?
                      this.getConsumedDataContent(voiceValue, translate('BILLS_USAGE_TALK'), aggregatedTotalVoiceQuota, translate('BILLS_USAGE_MIN'), isPrepaid, isUnlimitedVoice) :
                      <CircularProgress fill={voicePercentageConsumed} tintColor={colors.SECONDARY_TMOVE}>
                        {this.getConsumedDataContent(voiceValue, translate('BILLS_USAGE_TALK'), aggregatedTotalVoiceQuota, translate('BILLS_USAGE_MIN'), isPrepaid, isUnlimitedVoice)}
                      </CircularProgress>
                    : null
                }

                {hasVoice && hasData ? <View style={styles.separator} /> : null}

                {
                  hasData ?
                    unlimitedData || isPrepaid ?
                      this.getConsumedDataContent(dataValue, translate('BILLS_USAGE_DATA'), aggregatedTotalDataQuota, dataUnit, isPrepaid, unlimitedData) :
                      <CircularProgress fill={dataPercentageConsumed} tintColor={colors.SECONDARY_TMOVE}>
                        {this.getConsumedDataContent(dataValue, translate('BILLS_USAGE_DATA'), aggregatedTotalDataQuota, dataUnit, isPrepaid, unlimitedData)}
                      </CircularProgress>
                    : null
                }

              </View> : null
          }

          {
            hasHotExtraPackage ?
              <HotExtraPackageCard header={translate('BILLS_USAGE__HEP_TITLE')} caretPosition={hotExtraPackageCaretPosition} planValue={hotExtraPackage.planValue} planDescription={hotExtraPackage.planDescription} noOfDays={hotExtraPackage.days} dayDescription={translate('BILLS_USAGE__HEP_DAYS')} cost={hotExtraPackage.cost} costDescription={translate('BILLS_USAGE__HEP_COST_UNIT')} buttonText={translate('BILLS_USAGE__HEP_PAY_BUTTON')} />
              : null
          }
        </View>
        <Button type='inline' textType='BOLD' text={translate('BILLS_USAGE_SEE_OTHER_PACKAGE')} style={styles.otherButton} textStyle={styles.otherButtonText} onPress={navigateToBuyExtraPackage}/>

        {
          hasSharedPlan ?
            <Touchable style={styles.sharedTouchableWrapper} onPress={this.toggleModal(sharedNumbers, 'BILLS_USAGE_SHARED_NUMBER')}>
              <View style={styles.sharedNumbersContainer}>
                <ISText type='BOLD' style={styles.sharedNumbersDescription}>{translate('BILLS_USAGE_SHARED_NUMBER')}</ISText>
                <View style={styles.sharedNoIconWrap}><ISText type='BOLD' style={styles.sharedNumber}>{sharedNumbers.length}</ISText><Icon name='chevron-down' size={15} style={styles.sharedNumberIcon}/>
                </View>
              </View>
            </Touchable> : null
        }
        <Modal isVisible={isVisible} animationIn={'zoomIn'} animationOut={'zoomOut'}
          animationInTiming={1} backdropTransitionInTiming={1} backdropTransitionOutTiming={1}
          animationOutTiming={1} onBackdropPress={this.hideModal}>
          <Card header={translate(modalHeaderTextKey)} containerStyles={styles.modalContainer}>
            <List data={modalData} alignListItemsCenter onPress={this.selectItem} arrowVisibility={false}/>
          </Card>
          <Button onPress={this.toggleModal()} text={translate('STORE_LOCATOR__CANCEL')} style={styles.cancelButton}/>
        </Modal>
      </View>
    );
  }
}

CurrentUsageTM.defaultProps = {
  usageData: {
    AGGREGATED_QUOTA: {}
  },
  hotExtraPackage: null,
  navigateToBuyExtraPackage: noop,
  sharedNumbers: null,
  multiSimNumbers: null,
  isPrepaid: false
};

CurrentUsageTM.propTypes = {
  usageData: Proptypes.shape({
    MAIN_PACKAGE: Proptypes.array,
    EXTRA_PACKAGE: Proptypes.array
  }),
  hotExtraPackage: Proptypes.object,
  isPrepaid: Proptypes.bool,
  navigateToBuyExtraPackage: Proptypes.func,
  multiSimNumbers: Proptypes.array,
  sharedNumbers: Proptypes.array
};

export default CurrentUsageTM;
