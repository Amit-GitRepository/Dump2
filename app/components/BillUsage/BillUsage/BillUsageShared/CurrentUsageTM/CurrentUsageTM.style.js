import {colors, fonts} from '../../../../../themes/constants.styles';

export default {
  currentUsageContainer: {
    flexGrow: 1,
    paddingTop: 5
  },
  currentUsageWrapper: {
    paddingTop: 5,
    paddingHorizontal: 10
  },
  currentUsageSubContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-around',
    paddingHorizontal: 10,
    paddingBottom: 5
  },
  separator: {
    width: 10
  },
  circularProgress: {
    tintColor: colors.SECONDARY_TMOVE,
    width: 8,
    size: 120,
    backgroundColor: colors.PRIMARY_BG_SEPARATOR
  },
  circularProgressFillStyle: {
    fontSize: fonts.FONT_SIZE_XL,
    color: colors.SECONDARY_TMOVE
  },
  otherButtonText: {
    fontSize: fonts.FONT_SIZE_NORMAL
  },
  sharedNumbersContainer: {
    alignSelf: 'stretch',
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    paddingVertical: 10,
    paddingHorizontal: 10,
    borderTopWidth: 1,
    borderColor: colors.PRIMARY_BG_SEPARATOR
  },
  sharedNumbersDescription: {
    fontSize: fonts.FONT_SIZE_MEDIUM
  },
  sharedNumber: {
    fontSize: fonts.FONT_SIZE_MEDIUM,
    paddingRight: 10
  },
  sharedNumberIcon: {
    transform: [{rotate: '-90deg'}],
    color: colors.PRIMARY_ACTIONABLE,
    backgroundColor: colors.TRANSPARENT
  },
  sharedTouchableWrapper: {
    alignSelf: 'stretch'
  },
  sharedNoIconWrap: {
    flexDirection: 'row'
  },
  multiSimContainer: {
    alignItems: 'center',
    borderBottomWidth: 1,
    paddingBottom: 10,
    borderColor: colors.PRIMARY_BG_SEPARATOR,
    marginBottom: 10
  },
  multiSimText: {
    fontSize: fonts.FONT_SIZE_MEDIUM
  },
  modalContainer: {
    flex: 0,
    paddingBottom: 10,
    marginBottom: 5
  },
  detailsContainer: {
    flexDirection: 'column',
    alignItems: 'center'
  },
  infoBottomText: {
    fontSize: fonts.FONT_SIZE_SMALL,
    color: colors.PRIMARY_TEXT_TAB_LABEL,
    marginBottom: -5, // TODO: Remove after fonts update
    paddingTop: 2
  },
  infoTopText: {
    fontSize: fonts.FONT_SIZE_MEDIUM,
    color: colors.PRIMARY_TEXT_TAB_LABEL,
    marginBottom: -5, // TODO: Remove after fonts update
    marginTop: -10 // TODO: Remove after fonts update
  },
  infoSubText: {
    fontSize: fonts.FONT_SIZE_SMALL,
    color: colors.PRIMARY_TEXT_TAB_LABEL
  },
  valueText: {
    fontSize: fonts.FONT_SIZE_XL,
    marginBottom: -10, // TODO: Remove after fonts update
    marginTop: -5 // TODO: Remove after fonts update
  },
  prepaidContainer: {
    paddingTop: 10
  }
};
