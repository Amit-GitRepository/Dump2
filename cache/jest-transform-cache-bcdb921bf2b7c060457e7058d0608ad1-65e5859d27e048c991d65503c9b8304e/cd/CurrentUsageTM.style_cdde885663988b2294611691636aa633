Object.defineProperty(exports,"__esModule",{value:true});var cov_28jrytgc2g=function(){var path='/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/BillUsage/BillUsage/BillUsageShared/CurrentUsageTM/CurrentUsageTM.style.js',hash='a2ca9b4c1e98020c19413a2b9b84dfcc57209708',global=new Function('return this')(),gcv='__coverage__',coverageData={path:'/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/BillUsage/BillUsage/BillUsageShared/CurrentUsageTM/CurrentUsageTM.style.js',statementMap:{},fnMap:{},branchMap:{},s:{},f:{},b:{},_coverageSchema:'332fd63041d2c1bcb487cc26dd0d5f7d97098a6c'},coverage=global[gcv]||(global[gcv]={});if(coverage[path]&&coverage[path].hash===hash){return coverage[path];}coverageData.hash=hash;return coverage[path]=coverageData;}();var _constants=require('../../../../../themes/constants.styles');exports.default=

{
currentUsageContainer:{
flexGrow:1,
paddingTop:5},

currentUsageWrapper:{
paddingHorizontal:10},

currentUsageSubContainer:{
flexDirection:'row',
alignItems:'center',
justifyContent:'space-around',
paddingHorizontal:10,
paddingBottom:10},

separator:{
width:10},

circularProgress:{
tintColor:_constants.colors.SECONDARY_TMOVE,
width:8,
size:120,
backgroundColor:_constants.colors.PRIMARY_BG_SEPARATOR},

circularProgressFillStyle:{
fontSize:_constants.fonts.FONT_SIZE_XL,
color:_constants.colors.SECONDARY_TMOVE},

otherButtonText:{
fontSize:_constants.fonts.FONT_SIZE_NORMAL,
paddingTop:5},

sharedNumbersContainer:{
alignSelf:'stretch',
flexDirection:'row',
alignItems:'center',
justifyContent:'space-between',
paddingVertical:10,
paddingHorizontal:10,
borderTopWidth:1,
borderColor:_constants.colors.PRIMARY_BG_SEPARATOR},

sharedNumbersDescription:{
fontSize:_constants.fonts.FONT_SIZE_MEDIUM},

sharedNumber:{
fontSize:_constants.fonts.FONT_SIZE_MEDIUM,
paddingRight:10},

sharedNumberIcon:{
transform:[{rotate:'-90deg'}],
color:_constants.colors.PRIMARY_ACTIONABLE,
backgroundColor:_constants.colors.TRANSPARENT},

sharedTouchableWrapper:{
alignSelf:'stretch'},

sharedNoIconWrap:{
flexDirection:'row'},

multiSimContainer:{
alignItems:'center',
borderBottomWidth:1,
paddingBottom:10,
borderColor:_constants.colors.PRIMARY_BG_SEPARATOR,
marginBottom:10},

multiSimText:{
fontSize:_constants.fonts.FONT_SIZE_MEDIUM},

modalContainer:{
flex:0,
paddingBottom:10,
marginBottom:5},

detailsContainer:{
flexDirection:'column',
alignItems:'center'},

infoBottomText:{
fontSize:_constants.fonts.FONT_SIZE_SMALL,
color:_constants.colors.PRIMARY_TEXT_TAB_LABEL,
marginBottom:-5},

infoTopText:{
fontSize:_constants.fonts.FONT_SIZE_MEDIUM,
color:_constants.colors.PRIMARY_TEXT_TAB_LABEL,
marginBottom:-5,
marginTop:-10},

infoSubText:{
fontSize:_constants.fonts.FONT_SIZE_SMALL,
color:_constants.colors.PRIMARY_TEXT_TAB_LABEL},

valueText:{
fontSize:_constants.fonts.FONT_SIZE_XL,
marginBottom:-10,
marginTop:-5}};