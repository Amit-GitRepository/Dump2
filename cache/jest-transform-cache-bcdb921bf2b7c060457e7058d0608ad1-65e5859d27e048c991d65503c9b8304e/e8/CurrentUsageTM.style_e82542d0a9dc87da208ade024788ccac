Object.defineProperty(exports,"__esModule",{value:true});var cov_28jrytgc2g=function(){var path='/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/BillUsage/BillUsage/BillUsageShared/CurrentUsageTM/CurrentUsageTM.style.js',hash='a2ca9b4c1e98020c19413a2b9b84dfcc57209708',global=new Function('return this')(),gcv='__coverage__',coverageData={path:'/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/BillUsage/BillUsage/BillUsageShared/CurrentUsageTM/CurrentUsageTM.style.js',statementMap:{},fnMap:{},branchMap:{},s:{},f:{},b:{},_coverageSchema:'332fd63041d2c1bcb487cc26dd0d5f7d97098a6c'},coverage=global[gcv]||(global[gcv]={});if(coverage[path]&&coverage[path].hash===hash){return coverage[path];}coverageData.hash=hash;return coverage[path]=coverageData;}();var _constants=require('../../../../../themes/constants.styles');exports.default=

{
currentUsageContainer:{
flexGrow:1,
paddingTop:10,
paddingHorizontal:10},

currentUsageSubContainer:{
flexDirection:'row',
alignItems:'center',
justifyContent:'space-around',
paddingHorizontal:10,
paddingBottom:10},

separator:{
width:10},

circularProgress:{
tintColor:_constants.colors.SECONDARY_TMOVE,
width:8,
size:120,
backgroundColor:_constants.colors.PRIMARY_BG_SEPARATOR},

circularProgressFillStyle:{
fontSize:_constants.fonts.FONT_SIZE_XXL},

otherButtonText:{
fontSize:_constants.fonts.FONT_SIZE_NORMAL,
paddingTop:5},

sharedNumbersContainer:{
alignSelf:'stretch',
flexDirection:'row',
alignItems:'center',
justifyContent:'space-between',
paddingVertical:10,
borderTopWidth:1,
borderColor:_constants.colors.PRIMARY_BG_SEPARATOR},

sharedNumbersDescription:{
fontSize:_constants.fonts.FONT_SIZE_MEDIUM},

sharedNumber:{
fontSize:_constants.fonts.FONT_SIZE_MEDIUM,
paddingRight:10},

sharedNumberIcon:{
transform:[{rotate:'-90deg'}],
color:_constants.colors.PRIMARY_ACTIONABLE,
backgroundColor:_constants.colors.TRANSPARENT},

sharedTouchableWrapper:{
alignSelf:'stretch'},

sharedNoIconWrap:{
flexDirection:'row'}};