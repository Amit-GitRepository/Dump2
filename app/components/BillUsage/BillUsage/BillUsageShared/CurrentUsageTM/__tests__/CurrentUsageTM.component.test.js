import CurrentUsageTM from '../CurrentUsageTM.component';
import React from 'react'; // Enzyme adapter for React 16
import ReactSixteenAdapter from 'enzyme/build/adapters/ReactSixteenAdapter';
import renderer from 'react-test-renderer';
import {configure} from 'enzyme';

configure({adapter: new ReactSixteenAdapter()});

describe('CurrentUsageTM component', () => {
  it('renders correctly', () => {
    const component = renderer.create(<CurrentUsageTM />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
