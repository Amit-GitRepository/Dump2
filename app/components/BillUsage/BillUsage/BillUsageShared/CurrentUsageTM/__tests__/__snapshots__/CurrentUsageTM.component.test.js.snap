// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrentUsageTM component renders correctly 1`] = `
<View
  style={
    Object {
      "flexGrow": 1,
      "paddingTop": 5,
    }
  }
>
  <View
    style={
      Object {
        "paddingHorizontal": 10,
        "paddingTop": 5,
      }
    }
  />
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "transparent",
        "borderRadius": 50,
        "borderWidth": 0,
        "opacity": 1,
        "padding": 5,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "stretch",
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "color": "#303C46",
              },
              Array [
                Object {
                  "color": "#F93E60",
                  "fontSize": 16,
                },
                Object {
                  "fontSize": 16,
                },
              ],
              Object {
                "fontFamily": "SukhumvitSet-Bold",
              },
            ]
          }
        >
          See other extra packages
        </Text>
      </View>
    </View>
  </View>
  <Modal
    animationType="none"
    hardwareAccelerated={false}
    onModalHide={[Function]}
    onRequestClose={[Function]}
    transparent={true}
    visible={false}
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "black",
            "bottom": 0,
            "left": 0,
            "opacity": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "backgroundColor": "rgba(48,60,70,0.9)",
            "height": 1334,
            "width": 750,
          },
        ]
      }
      testID={undefined}
    />
    <View
      onModalHide={[Function]}
      pointerEvents="box-none"
      style={
        Array [
          Object {
            "margin": 37.5,
            "transform": Array [
              Object {
                "translateY": 0,
              },
            ],
          },
          Object {
            "flex": 1,
            "justifyContent": "center",
          },
          undefined,
        ]
      }
    >
      <View
        elevation={5}
        style={
          Array [
            Object {
              "backgroundColor": "#FFFFFF",
              "borderRadius": 8,
              "flex": 1,
              "marginHorizontal": -1,
              "overflow": "hidden",
              "shadowColor": "#303C46",
              "shadowOffset": Object {
                "height": 2,
                "width": 1,
              },
              "shadowOpacity": 0.3,
              "shadowRadius": 1,
            },
            Object {
              "flex": 0,
              "marginBottom": 5,
              "paddingBottom": 10,
            },
          ]
        }
      >
        <View
          onPress={null}
          style={
            Array [
              Object {
                "zIndex": 1,
              },
              Object {},
            ]
          }
        >
          <View>
            <View
              style={
                Object {
                  "borderWidth": 0,
                  "elevation": 3,
                  "shadowColor": "#303C46",
                  "shadowOffset": Object {
                    "height": 2,
                    "width": 1,
                  },
                  "shadowOpacity": 0.3,
                  "shadowRadius": 1,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "borderBottomColor": "#F93E60",
                      "paddingHorizontal": 5,
                    },
                    Object {
                      "borderBottomColor": "#303C46",
                    },
                  ]
                }
              >
                <View
                  style={
                    Array [
                      Array [
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "#E9E9EA",
                          "padding": 10,
                        },
                        Object {
                          "backgroundColor": "#303C46",
                          "marginHorizontal": -5,
                          "padding": 10,
                        },
                      ],
                      Object {},
                    ]
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "backgroundColor": "transparent",
                          "color": "#303C46",
                        },
                        Array [
                          Object {
                            "color": "#000000",
                            "fontSize": 18,
                          },
                          Object {
                            "color": "#FFFFFF",
                          },
                        ],
                        Object {
                          "fontFamily": "SukhumvitSet-Bold",
                        },
                      ]
                    }
                  >
                    [missing "en." translation]
                  </Text>
                </View>
              </View>
            </View>
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#303C46",
                    "fontSize": 24,
                  },
                  Array [
                    Object {
                      "alignSelf": "center",
                      "backgroundColor": "transparent",
                    },
                    Object {
                      "bottom": 9.6,
                      "marginBottom": -9.6,
                    },
                    Object {},
                  ],
                  Object {
                    "fontFamily": "iservice",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              Ó§Å
            </Text>
          </View>
        </View>
        <RCTScrollView
          data={Array []}
          disableVirtualization={false}
          getItem={[Function]}
          getItemCount={[Function]}
          horizontal={false}
          initialNumToRender={10}
          keyExtractor={[Function]}
          maxToRenderPerBatch={10}
          numColumns={1}
          onContentSizeChange={[Function]}
          onEndReachedThreshold={2}
          onLayout={[Function]}
          onMomentumScrollEnd={[Function]}
          onScroll={[Function]}
          onScrollBeginDrag={[Function]}
          onScrollEndDrag={[Function]}
          renderItem={[Function]}
          scrollEventThrottle={50}
          stickyHeaderIndices={Array []}
          updateCellsBatchingPeriod={50}
          viewabilityConfigCallbackPairs={Array []}
          windowSize={21}
        >
          <View />
        </RCTScrollView>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignSelf": "stretch",
            "backgroundColor": "#F93E60",
            "borderRadius": 50,
            "opacity": 1,
            "padding": 10,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "stretch",
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "color": "#303C46",
                  },
                  Array [
                    Object {
                      "color": "#FFFFFF",
                      "fontSize": 16,
                    },
                    Object {},
                  ],
                  Object {
                    "fontFamily": "SukhumvitSet-Bold",
                  },
                ]
              }
            >
              Cancel
            </Text>
          </View>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;
